lib/internal/util/inspect.js

```diff
@@ -1320,12 +1323,50 @@ function markCwd(ctx, line, workingDirectory) {
   return tempLine;
 }
 
-function safeGetCWD() {
+// The following characters are percent-encoded when converting from file path
+// to URL:
+// - %: The percent character is the only character not encoded by the
+//        `pathname` setter.
+// - \: Backslash is encoded on non-windows platforms since it's a valid
+//      character but the `pathname` setters replaces it by a forward slash.
+// - LF: The newline character is stripped out by the `pathname` setter.
+//       (See whatwg/url#419)
+// - CR: The carriage return character is also stripped out by the `pathname`
+//       setter.
+// - TAB: The tab character is also stripped out by the `pathname` setter.
+const percentRegEx = /%/g;
+const backslashRegEx = /\\/g;
+const newlineRegEx = /\n/g;
+const carriageReturnRegEx = /\r/g;
+const tabRegEx = /\t/g;
+
+function encodePathChars(filepath) {
+  if (StringPrototypeIncludes(filepath, '%'))
+    filepath = StringPrototypeReplace(filepath, percentRegEx, '%25');
+  // In posix, backslash is a valid character in paths:
+  if (!isWindows && StringPrototypeIncludes(filepath, '\\'))
+    filepath = StringPrototypeReplace(filepath, backslashRegEx, '%5C');
+  if (StringPrototypeIncludes(filepath, '\n'))
+    filepath = StringPrototypeReplace(filepath, newlineRegEx, '%0A');
+  if (StringPrototypeIncludes(filepath, '\r'))
+    filepath = StringPrototypeReplace(filepath, carriageReturnRegEx, '%0D');
+  if (StringPrototypeIncludes(filepath, '\t'))
+    filepath = StringPrototypeReplace(filepath, tabRegEx, '%09');
+  return filepath;
+}
+
+function safeGetEscapedCWD(stackFramePart) {
+  let workingDirectory;
   try {
-    return process.cwd();
+    workingDirectory = process.cwd();
   } catch {
-    // Explicitly empty
+    return;
+  }
+  // Handle ESM
+  if (stackFramePart.includes('file://')) {
+    return `file://${encodePathChars(workingDirectory)}`;
```

Comment:
Instead of moving `encodePathChars` in this file, shouldn't we call `pathToFileURL`?

```suggestion
    return pathToFileURL(workingDirectory);
```

Reply:
It is slower as it would create a URL instance and then do some transformations. Just escaping the input should already be enough. We know that it's always `file://` (besides theoretically an UNC path). This is a best effort under all circumstances. Thus, I think it's fine to use it like this.

Reply:
I just checked again and there is indeed need to use pathToFileUrl, so I just changed it to it and made sure the test also checks for a character that is not in encodePathChars.